src/rf/apps/workers/process.py

```diff
@@ -252,16 +250,20 @@ def emr_heartbeat(self, record):
             return False
 
         try:
-            int(layer_id)
+            layer_id = int(layer_id)
         except ValueError:
             return False
 
         try:
             layer = Layer.objects.get(id=layer_id)
         except Layer.DoesNotExist:
-            return False
+            return True
+
+        log.debug('Heartbeat for layer %d', layer_id)
 
-        log.info('Heartbeat for layer %d', layer_id)
+        if layer.status in (enums.STATUS_COMPLETED, enums.STATUS_FAILED):
+            log.debug('Ending heartbeat')
```

Comment:
Consider making this more explicit:
`log.debug('Ending heartbeat. Job is complete.')`